app/components/Sidebar/components/StarredLink.tsx

```diff
@@ -40,25 +40,35 @@ function StarredLink({ star }: Props) {
   const { documentId, collectionId } = star;
   const collection = collections.get(collectionId);
   const locationSidebarContext = useLocationState();
-  const sidebarContext = useSidebarContext();
+  const starType = star.documentId ? "document" : "collection";
+  const sidebarContext = starredSidebarContext(
+    starType === "document" ? star.documentId! : star.collectionId
+  );
   const [expanded, setExpanded] = useState(
-    star.collectionId === ui.activeCollectionId &&
+    (starType === "document"
+      ? star.documentId === ui.activeDocumentId
+      : star.collectionId === ui.activeCollectionId) &&
       sidebarContext === locationSidebarContext
   );
 
   React.useEffect(() => {
-    if (
-      star.documentId === ui.activeDocumentId &&
-      sidebarContext === locationSidebarContext
-    ) {
-      setExpanded(true);
-    } else if (
-      star.collectionId === ui.activeCollectionId &&
-      sidebarContext === locationSidebarContext
-    ) {
-      setExpanded(true);
+    if (starType === "document") {
```

Comment:
I don't really know that this addition is neccessary?

Reply:
Not really, added for readability - will remove it.